src/crimson/os/seastore/seastore.cc

```diff
@@ -2883,4 +2494,397 @@ std::unique_ptr<SeaStore> make_test_seastore(
+    }
+  });
+}
+
+SeaStore::base_iertr::future<>
+SeaStore::Shard::omaptree_rm_keys(
+  Transaction& t,
+  omap_root_t&& root,
+  Onode& onode,
+  omap_keys_t&& keys)
+{
+  LOG_PREFIX(SeaStoreS::omaptree_rm_keys);
+  auto type = root.get_type();
+  if (root.is_null()) {
+    DEBUGT("{} {} keys, null root", t, type, keys.size());
+    return seastar::now();
+  }
+  DEBUGT("{} {} keys ...", t, type, keys.size());
+  return seastar::do_with(
+    BtreeOMapManager(*transaction_manager),
+    std::move(root),
+    std::move(keys),
+    [&t, &onode, FNAME, type]
+    (auto &omap_manager, auto &root, auto &keys)
+  {
+    return trans_intr::do_for_each(
+      keys.begin(),
+      keys.end(),
+      [&omap_manager, &t, &root, FNAME, type](auto &p)
+    {
+      DEBUGT("{} remove key={} ...", t, type, p);
+      return omap_manager.omap_rm_key(root, t, p);
+    }).si_then([&t, &root, &onode] {
+      if (root.must_update()) {
+        omaptree_update_root(t, root, onode);
+      }
+    });
+  });
+}
+
+SeaStore::base_iertr::future<>
+SeaStore::Shard::omaptree_rm_keyrange(
+  Transaction& t,
+  omap_root_t&& root,
+  Onode& onode,
+  std::string first,
+  std::string last)
+{
+  LOG_PREFIX(SeaStoreS::omaptree_rm_keyrange);
+  auto type = root.get_type();
+  if (first > last) {
+    ERRORT("{} {} > {}, range error", t, type, first, last);
+    ceph_abort();
+  }
+  if (root.is_null()) {
+    DEBUGT("{} {} ~ {}, null root", t, type, first, last);
+    return seastar::now();
+  }
+  DEBUGT("{} {} ~ {} ...", t, type, first, last);
+  return seastar::do_with(
+    BtreeOMapManager(*transaction_manager),
+    std::move(root),
+    std::move(first),
+    std::move(last),
+    [&t, &onode, FNAME]
+    (auto &omap_manager, auto &root, auto &first, auto &last)
+  {
+    auto config = OMapManager::omap_list_config_t()
+      .with_inclusive(true, false)
+      .without_max();
+    return omap_manager.omap_rm_key_range(
+      root, t, first, last, config
+    ).si_then([&t, &root, &onode, FNAME] {
+      if (root.must_update()) {
+        omaptree_update_root(t, root, onode);
+      }
+      DEBUGT("done", t);
+    });
+  });
+}
+
+SeaStore::base_iertr::future<>
+SeaStore::Shard::omaptree_rm_key(
+  Transaction& t,
+  omap_root_t&& root,
+  Onode& onode,
+  std::string&& name)
+{
+  LOG_PREFIX(SeaStoreS::omaptree_rm_key);
+  if (root.is_null()) {
+    DEBUGT("{} key={}, null root", t, root.get_type(), name);
+    return base_iertr::now();
+  }
+  DEBUGT("{} key={} ...", t, root.get_type(), name);
+  return seastar::do_with(
+    BtreeOMapManager(*transaction_manager),
+    std::move(root),
+    std::move(name),
+    [&t, &onode, FNAME](auto &omap_manager, auto &root, auto &name)
+  {
+    return omap_manager.omap_rm_key(root, t, name
+    ).si_then([&t, &root, &onode, &name, FNAME] {
+      if (root.must_update()) {
+        omaptree_update_root(t, root, onode);
+      }
+      DEBUGT("{} key={} removed", t, root.get_type(), name);
+    });
+  });
+}
+
```

Comment:
I didn't check line-by-line. I suppose this commit should only move the code.

Reply:
This commit simply reorganizes the code to group the omap-related implementation, like your commit.